savings-account/src/lib.rs

```diff
@@ -2,8 +2,390 @@
+        Ok(())
+    }
+
+    // endpoints
+
+    #[payable("*")]
+    #[endpoint]
+    fn deposit(
+        &self,
+        #[payment_token] payment_token: TokenIdentifier,
+        #[payment_amount] payment_amount: Self::BigUint,
+    ) -> SCResult<()> {
+        self.require_lend_token_issued()?;
+
+        let stablecoin_token_id = self.stablecoin_token_id().get();
+        require!(
+            payment_token == stablecoin_token_id,
+            "May only deposit stablecoins"
+        );
+
+        let lend_token_id = self.lend_token_id().get();
+        let sft_nonce = self.create_tokens(&lend_token_id, &payment_amount, &())?;
+
+        self.lend_metadata(sft_nonce).set(&LendMetadata {
+            timestamp: self.blockchain().get_block_timestamp(),
+            amount_in_circulation: payment_amount.clone(),
+        });
+
+        let caller = self.blockchain().get_caller();
+        self.send()
+            .direct(&caller, &lend_token_id, sft_nonce, &payment_amount, &[]);
+
+        Ok(())
+    }
+
+    #[payable("*")]
+    #[endpoint]
+    fn borrow(
+        &self,
+        #[payment_token] payment_token: TokenIdentifier,
+        #[payment_nonce] payment_nonce: u64,
+        #[payment_amount] payment_amount: Self::BigUint,
+    ) -> SCResult<()> {
+        self.require_borrow_token_issued()?;
+
+        let liquid_staking_token_id = self.liquid_staking_token_id().get();
+        require!(
+            payment_token == liquid_staking_token_id,
+            "May only use liquid staking position as collateral"
+        );
+
+        let stablecoin_token_id = self.stablecoin_token_id().get();
+        let stablecoin_reserves = self.get_reserves(&stablecoin_token_id);
+        let borrowed_amount = self.borrowed_amount().get();
+
+        let pool_params = self.pool_params().get();
+        let current_utilisation =
+            self.compute_capital_utilisation(&borrowed_amount, &stablecoin_reserves);
+        let borrow_rate = self.compute_borrow_rate(
+            &pool_params.base_borrow_rate,
+            &pool_params.borrow_rate_under_opt_factor,
+            &pool_params.borrow_rate_over_opt_factor,
+            &pool_params.optimal_utilisation,
+            &current_utilisation,
+        );
+
+        let egld_price_in_stablecoin = self.get_egld_price_in_stablecoin()?;
+        let staking_position_value =
+            self.compute_staking_position_value(&egld_price_in_stablecoin, &payment_amount);
+        let borrow_value = self.compute_borrow_amount(&borrow_rate, &staking_position_value);
+
+        require!(borrow_value > 0, "Deposit amount too low");
+
+        let sc_stablecoin_balance = self.blockchain().get_sc_balance(&stablecoin_token_id, 0);
+        require!(
+            borrow_value <= sc_stablecoin_balance,
+            "Not have enough funds to lend"
+        );
+
+        let borrow_token_id = self.borrow_token_id().get();
+        let borrow_token_nonce = self.create_tokens(&borrow_token_id, &payment_amount, &())?;
```

Comment:
so the newly created token have only Nonce and Value - all the rest is kept in the stoage of Nonce-Attributes ?

Reply:
Yep, the metadata is stored here:
```
#[storage_mapper("borrowMetadata")]
    fn borrow_metadata(
        &self,
        sft_nonce: u64,
    ) -> SingleValueMapper<Self::Storage, BorrowMetadata<Self::BigUint>>;
```

The Rust framework Nonce-Attributes mapper is not released yet, so I'm using my own, but will probably update to that one on the next version.